tests/geometry/test_geometry_property_checks.py

```diff
@@ -0,0 +1,406 @@
+import numpy as np
```

Comment:
Consider grouping it into dedicated classes and moving dense arrays to `reference_dense_arrays.py`